esx_service/vmdk_ops_test.py

```diff
@@ -252,7 +255,156 @@ def test_failure(self):
             with self.assertRaises(vmdk_ops.ValidationError):
                 vmdk_ops.validate_opts(opts, self.path)
 
+    
+class VmdkAttachDetachTestCase(unittest.TestCase):
+    """ Unit test for VMDK Attach and Detach ops """
 
+    volNamePre = "vol_UnitTest_Attach"
+    vm_name = 'test-vm'
+    max_vol_count = 60
+    
+    def setUp(self):
+        """ Setup run before each test """
+        logging.debug("VMDKAttachDetachTest setUp path =%s", path)
+        self.cleanup()
+        
+        writeable_ds_path = 'datastore1'
+        if writeable_ds_path:
+            # get service_instance, and create a VM
+            if not vmdk_ops.si:
+                vmdk_ops.connectLocal()
+
+            self.create_vm(vmdk_ops.si, self.vm_name, writeable_ds_path)
+
+            # create max_vol_count+1 VMDK files
+            for id in range(1, self.max_vol_count+2):
+                volName = 'VmdkAttachDetachTestVol' + str(id)
+                fullpath = os.path.join(path, volName + '.vmdk')
```

Comment:
Using the global path is likely incorrect here. It may point to a vsan datastore after other tests run and not the datastore selected above. Ideally this file wouldn't contain any  globals at all, but that's not your fault. You can instead use the dockvol-path returned from the `vmdk_utils.get_datastores()` function I suggested you call above.